sklearn/preprocessing/_function_transformer.py

```diff
@@ -200,7 +200,9 @@ def _check_inverse_transform(self, X):
             # Dataframes can have multiple dtypes
             dtypes = X.dtypes
 
-        if not all(np.issubdtype(d, np.number) for d in dtypes):
+        if not all(
+            isinstance(d, np.dtype) and np.issubdtype(d, np.number) for d in dtypes
+        ):
```

Comment:
This currently also does not support the pandas numeric extension dtypes (for which there is some support in eg `check_array`)